rust/lance-index/src/scalar/inverted/index.rs

```diff
@@ -2390,17 +2446,42 @@ pub fn flat_bm25_search_stream(
     input: SendableRecordBatchStream,
     doc_col: String,
     query: String,
-    index: &InvertedIndex,
+    index: &Option<InvertedIndex>,
+    unindexed_scorer: MemoryBM25Scorer,
 ) -> SendableRecordBatchStream {
-    let mut tokenizer = Box::new(index.tokenizer.clone());
+    let mut tokenizer = match index {
+        Some(index) => index.tokenizer(),
+        // TODO: allow users to specify a tokenizer when querying columns without an inverted index.
```

Comment:
I think creating an index is how they should specify the tokenizer. Even if they just pass `train=False`, we at least get the configuration.